tools/k-rule-find/main.cpp

```diff
@@ -0,0 +1,121 @@
+#include "kllvm/ast/AST.h"
+#include "kllvm/parser/KOREParser.h"
+#include "llvm/Support/CommandLine.h"
+
+#include <dlfcn.h>
+#include <iostream>
+#include <stdlib.h>
+
+#include "runtime/header.h"
+
+using namespace llvm;
+using namespace kllvm;
+
+struct Location {
+  char *filename;
+  int64_t line;
+  int64_t column;
+};
+
+std::string SOURCE_ATT = "org'Stop'kframework'Stop'attributes'Stop'Source";
+std::string LOCATION_ATT = "org'Stop'kframework'Stop'attributes'Stop'Location";
+
+cl::OptionCategory KRuleCat("k-rule-find options");
+
+cl::opt<std::string> KompiledDir(
+    cl::Positional, cl::desc("<kompiled-dir>"), cl::Required,
+    cl::cat(KRuleCat));
+
+cl::opt<std::string> RuleLocation(
+    cl::Positional, cl::desc("<filename.k:line[:column]>"), cl::Required,
+    cl::cat(KRuleCat));
+
+Location parseLocation(std::string loc) {
+  char *filename;
```

Comment:
Why does this need to be declared first?

Reply:
No, do you want it to be declared after `getSource` and `getLocation`?

Reply:
I mean the variable `filename` - in general leaving things uninitialised isn't best practice if you can declare + initialise at the same time.

Reply:
Aah, okok!